### PR TITLE
MTDSA-13562: TYS Updates For Trigger Calculation

### PR DESCRIPTION
**Pull request by tom-lloyd-hmrc**

### CODE REVIEW

File: resources/public/api/conf/3.0/errors.raml [hunk 1]
```diff
@@ -192,13 +192,79 @@ traits:
...
+      400:
+         body:
+           application/json:
+             type: types.errorResponse
+             examples:
+               ruleTaxYearNotEnded:
+                 description: 'A calculation with finalDeclaration true cannot be triggered before the taxYear to which it relates has ended. This error only occurs for taxYears from 2023-24 onwards.'
+                 value:
+                   code: RULE_TAX_YEAR_NOT_ENDED
+  ruleRecentSubmissionsExistTys:
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply resources/public/api/conf/3.0/errors.raml]: Good point, renamed.

File: resources/public/api/conf/3.0/errors.raml [hunk 2]
```diff
@@ -192,13 +192,79 @@ traits:
...
+    responses:
+      400:
+         body:
+           application/json:
+             type: types.errorResponse
+             examples:
+               ruleRecentSubmissionsExist:
+                 description: 'More recent submissions exist. You must first trigger a calculation with finalDeclaration false. This error only occurs for taxYears from 2023-24 onwards.'
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply resources/public/api/conf/3.0/errors.raml]: I've updated to this format, although it doesn't seem to have any impact on the formatting in the browser. 
image
Not sure if I've done something wrong?

File: it/v3/endpoints/TriggerCalculationControllerISpec.scala [hunk 3]
```diff
@@ -137,16 +102,16 @@ class TriggerCalculationControllerISpec extends V3IntegrationBaseSpec {
            """.stripMargin
 
         def serviceErrorTest(backendStatus: Int, backendCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply it/v3/endpoints/TriggerCalculationControllerISpec.scala:119]: Woops, good spot!

File: it/v3/endpoints/TriggerCalculationControllerISpec.scala [hunk 4]
```diff
@@ -170,4 +135,65 @@ class TriggerCalculationControllerISpec extends V3IntegrationBaseSpec {
...
+  private trait Test {
+
+    val nino             = "AA123456A"
+    val finalDeclaration = true
+
+    def mtdTaxYear: String
+    def downstreamUri: String
+    def setupStubs(): StubMapping
+
+    def request(nino: String, taxYear: String, maybeFinalDeclaration: Option[Boolean]): WSRequest = {
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply it/v3/endpoints/TriggerCalculationControllerISpec.scala:171]: Added a scenario to the parser spec and the integration spec to cover this.

File: resources/public/api/conf/3.0/errors.raml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremystone) [reply resources/public/api/conf/3.0/errors.raml]: Oh well. The source is more readable now anyway. Looks like this approach works for queryParameters (and some other descriptions) but not for errors 🤷 .

**Review by jeremystone — APPROVED** [review]

**Pull request merged**